### PR TITLE
Stub out API and base directory for ObjectiveC interop tests

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Stub out the API and the infra for the tests. Add the APIs to the System.Runtime.InteropServices ref assembly.

### CODE REVIEW

File: src/tests/Interop/ObjectiveC/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+set(CMAKE_OBJC_STANDARD 11)
```
Comment (elinor-fung) [inline src/tests/Interop/ObjectiveC/CMakeLists.txt:1]: Should we also set the minimum cmake version? You had mentioned that the built-in Objective-C support required something higher than our current official minimum.

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/Wrappers.PlatformNotSupported.cs [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+        /// is the Objective-C instance.
+        /// <code>
+        /// int ref_callback(void* id)
+        /// {
+        ///    return 0; // Return zero for no reference or 1 for reference
+        /// }
+        /// </code>
+        /// </remarks>
+        /// <see cref="RegisterInstanceFlags.ManagedDefinition"/>
+        protected unsafe virtual delegate* unmanaged[Cdecl]<IntPtr, int> GetReferenceCallback(bool isManagedRegistration) =>
```
Comment (elinor-fung) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/Wrappers.PlatformNotSupported.cs:154]: Duplicated `=>`

File: src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems [hunk 3]
```diff
@@ -1963,4 +1963,14 @@
...
   </ItemGroup>
+  <ItemGroup Condition="'$(FeatureObjcWrappers)' == 'true'">
+    <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\ObjectiveC\Flags.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\ObjectiveC\ManagedObjectWrapperLifetime.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\ObjectiveC\Instance.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\ObjectiveC\ObjectiveCBase.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\ObjectiveC\Wrappers.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(FeatureObjcWrappers)' != 'true'">
+    <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\ObjectiveC\Wrappers.PlatformNotSupported.cs" />
```
Comment (elinor-fung) [inline src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems:1974]: Don't we still need the flags to be defined?

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/ObjectiveCBase.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace System.Runtime.InteropServices.ObjectiveC
+{
+    /// <summary>
+    /// Base type for all types participating in Objective-C interop.
+    /// </summary>
+    [SupportedOSPlatform("macos")]
+    public abstract class ObjectiveCBase : IDisposable
+    {
+        public static readonly IntPtr InvalidInstanceValue = (IntPtr)(-1);
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/ObjectiveCBase.cs:15]: Let's get rid of this. I forgot to delete it.

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/Wrappers.PlatformNotSupported.cs [hunk 5]
```diff
@@ -0,0 +1,299 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Runtime.Versioning;
+
+namespace System.Runtime.InteropServices.ObjectiveC
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/Wrappers.PlatformNotSupported.cs:7]: We should remove doc from the "not supported" impl.

**Review by AaronRobinsonMSFT — COMMENTED** [review]
LGTM

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/Wrappers.PlatformNotSupported.cs (hunk 5 above)
Comment (jkoritzinsky) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/Wrappers.PlatformNotSupported.cs:7]: I'm worried that if we remove the PNSE impl's docs, the API docs won't automatically be ported.  I don't know which build the docs team is planning on using to harvest docs from.

File: src/libraries/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs [hunk 6]
```diff
@@ -1796,6 +1796,127 @@ public enum VARKIND
...
+    }
+
+    [System.Runtime.Versioning.SupportedOSPlatformAttribute("macos")]
+    public abstract class ObjectiveCBase : IDisposable
+    {
+        public static readonly IntPtr InvalidInstanceValue = (IntPtr)(-1);
+
+        protected ObjectiveCBase() { }
+
+        /// <inheritdoc />
```
Comment (elinor-fung) [inline src/libraries/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs:1866]: Remove comment and new lines (I think the ref files usually don't have empty lines).

**Review by elinor-fung — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/Wrappers.PlatformNotSupported.cs (hunk 5 above)
Comment (jkoritzinsky) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ObjectiveC/Wrappers.PlatformNotSupported.cs:7]: I'll leave this for now until we are sure that we can remove them and still get the docs auto-ported.

**Pull request merged**